waflib/Task.py

```diff
@@ -411,10 +411,11 @@ def format_error(self):
 		elif not self.hasrun:
 			return 'task in %r was not executed for some reason: %r' % (name, self)
 		elif self.hasrun == CRASHED:
+			msg = ' '.join(msg)
 			try:
-				return ' -> task in %r failed (exit status %r): %r\n%r' % (name, self.err_code, self, msg)
+				return ' -> task in %r failed (exit status %r): %r\n%s' % (name, self.err_code, self, msg)
 			except AttributeError:
-				return ' -> task in %r failed: %r\n%r' % (name, self, msg)
+				return ' -> task in %r failed: %r\n%s' % (name, self, msg)
 		elif self.hasrun == MISSING:
 			return ' -> missing files in %r: %r\n%r' % (name, self, msg)
```

Comment:
I didn't really investigate in what cases it would fall in this. I thought the "msg" here would not be coming from a compile command, but rather a msg from waf itself. In this case I didn't want to break other things and since the user isn't supposed to copy and paste to reproduce the build of a single compilation unit, I didn't feel inclined to take the risk.